fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java

```diff
@@ -3281,12 +3257,152 @@ private static class HorizontalPaymentAllocationContext implements LoopContext {
     private boolean isInterestRecalculationSupported(TransactionCtx ctx, Loan loan) {
         if (ctx instanceof ProgressiveTransactionCtx progressiveTransactionCtx) {
             return loan.isInterestBearingAndInterestRecalculationEnabled() && !progressiveTransactionCtx.isChargedOff()
-                    && !progressiveTransactionCtx.isWrittenOff() && !progressiveTransactionCtx.isContractTerminated();
+                    && !progressiveTransactionCtx.isWrittenOff() && !progressiveTransactionCtx.isContractTerminated()
+                    && keepUsingEmiCalculatorBasedOnReAge(progressiveTransactionCtx);
         } else {
             return false;
         }
     }
 
+    private void handleReAgeWithCommonStrategy(LoanTransaction loanTransaction, CommonReAgeSettings settings, TransactionCtx ctx) {
```

Comment:
This should be part of the calculator and the model, not the transaction processor.